components/CheckboxInput.tsx

```
@@ -0,0 +1,57 @@
+        "w-4 h-4 border-2 border-gray-400 transition-colors relative",
+        {
+          "bg-transparent": state === 0,
+          "bg-gray-400": state === 2,
+          "cursor-pointer hover:border-gray-600": !!onChange,
+          "cursor-default": !onChange,
+        },
+        className
+      )}
+      aria-label={`Toggle ${id}`}
+      aria-checked={state === 2}
+      role="checkbox"
```

Biome

[error] The elements with the following roles can be changed to the following elements:
<input type="checkbox">


For examples and more information, see WAI-ARIA Roles

(lint/a11y/useSemanticElements)